src/glue.c

```
@@ -0,0 +1,203 @@
+#include "verify.h"
+#include <node_api.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+//#define VF_DEBUG 1
+
+napi_status HandleError(napi_env env, struct Error *err) {
+  napi_status status;
+  napi_value error;       // The js Error object
+  napi_value errors;      // The js list of error strings
+  napi_value errorString; // Used to build the list of strings
+
+  char *msg = NULL; // used to build strings
+
+  if (err == NULL) {
+    status =
+        napi_throw_error(env, NULL, "Unknown exception verifying document");
+    return status;
+  }
+
+  // Create an array for all of the error strings
+  status = napi_create_array(env, &errors);
+  if (status != napi_ok)
+    return status;
+
+#ifdef VF_DEBUG
+  printf("created js array\n");
+#endif
+
+  // We're going to create the error object so that the head of the err linked
+  // list is the error object's message.  This *should* be the root error
+  msg = VF_err_fmt(err);
+  status = napi_create_string_utf8(env, msg, NAPI_AUTO_LENGTH, &errorString);
+#ifdef VF_DEBUG
+  printf("created js string for Error.message: %s\n", msg);
+#endif
+  free(msg);
+  msg = NULL;
+
+  if (status != napi_ok)
+    return status;
+
+  status = napi_create_error(env, NULL, errorString, &error);
+  if (status != napi_ok)
+    return status;
+
+#ifdef VF_DEBUG
+  printf("created js Error\n");
+#endif
+
+  // We're limiting to the first 1000 so that we don't
```

1000 (here) or 10 (below)?  It would probably be good to indicate that errors were dropped when hitting the limit (for example, adding "..subsequent errors omitted").

Also, `VF_err_free` will hang forever if this is a circular linked-list, so I'm not sure this check helps..

---

I've replaced the numbers with a macro `MAX_OPENSSL_ERRORS` for the display and set it to 25.  The difference is that `VF_err_free` is required in order to avoid a memory leak.  It's also something defined in the same file, and building the list is tested in the unit tests.

---

I've added a specific unit test to ensure that the list is built and consumed correctly